src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditorCore.jsx

```diff
@@ -0,0 +1,239 @@
+    allRelationships,
+    currentCategory,
+    [],
+  );
+  const relationshipsWithoutTarget = targetRelationshipCategory.filter(
+    entry => entry !== relationship,
+  );
+  allRelationships[currentCategory] = relationshipsWithoutTarget;
+  return allRelationships;
+}
+
+function updateRelationshipInCategory(
+  allRelationships,
+  category,
+  oldRelationshipName,
+  newRelationshipName,
+) {
+  const currentRelationships = get(allRelationships, category);
+
+  const oldRelationshipIndex = currentRelationships?.indexOf(
+    oldRelationshipName,
+  );
+  if (oldRelationshipIndex > -1)
+    currentRelationships[oldRelationshipIndex] = newRelationshipName;
+  allRelationships[category] = currentRelationships;
+  return allRelationships;
+}
+
+function createCategory(allRelationships) {
+  return { ...allRelationships, ['']: [] };
+}
+
+function updateCategoryName(
+  allRelationships,
+  currentCategoryName,
+  newCategoryName,
+) {
+  const relationshipNames = get(
+    allRelationships,
+    currentCategoryName,
+  );
+  delete allRelationships[currentCategoryName];
+  allRelationships[newCategoryName] = relationshipNames;
+  return allRelationships;
+}
+
+function addRelationshipToCategory(
+  allRelationships,
+  currentCategory,
+  newRelationship,
+) {
+  const currentRelationships = get(
+    allRelationships,
+    currentCategory,
+    [],
+  );
+  let newRelationships = currentRelationships;
+  if (!currentRelationships.includes(newRelationship))
+    newRelationships = [...currentRelationships, newRelationship];
+  allRelationships[currentCategory] = newRelationships;
+  return allRelationships;
+}
+
+function renderSingleRelationship(
+  allRelationships,
+  currentCategory,
+  relationship,
+  onChange,
+) {
+  return (
+    <div style={{ marginLeft: 32, marginTop: 10 }} key={relationship}>
+      <TextInput
+        width={240}
+        schema={relationship}
+        onChange={newRelationshipName => {
+          onChange(
+            updateRelationshipInCategory(
+              allRelationships,
+              currentCategory,
+              relationship,
+              newRelationshipName,
+            ),
+          );
+        }}
+        value={relationship}
+        autoFocus
+        InputProps={{
+          endAdornment: (
+            <InputAdornment position="end">
+              <DeleteButton
+                onClick={() => {
+                  onChange(
+                    deleteRelationship(
+                      allRelationships,
+                      currentCategory,
+                      relationship,
+                    ),
+                  );
+                }}
+              />
+            </InputAdornment>
+          ),
+        }}
+      />
+    </div>
+  );
+}
+
+function renderSingleRelationshipObj(
+  allRelationships,
+  targetKey,
+  onChange,
+) {
+  const relationships = get(allRelationships, targetKey, []);
+  return (
+    <div style={{ marginTop: 10 }} key={targetKey}>
```

Comment:
When `renderSingleRelationshipObj` is called as a React component (mentioned in another comment), this div will no longer need a key.